### PR TITLE
Give the user ability to opt out of get balance batch request for all loaded accounts

### LINKED ISSUE #15169: [Bug]: Don't correlate accounts with RPC endpoint.
### Describe the bug

When first unlocking the wallet, MetaMask sends a batch balance request for every account to the currently selected JSON-RPC endpoint.  This means that the JSON-RPC endpoint can link all of the user's accounts.  This also means that if a user *accidentally* connects to the wrong JSON-RPC endpoint it also leaks all of their addresses (not just the selected one) to that endpoint.

This is true also for other chains, as a user connecting to a test network or L2 via MM will also send the RPC provider for that chain all of their accounts rather than just the selected account.

### Steps to reproduce

1. Create a wallet with multiple accounts.
2. Sign-in to MM.
3. Notice that MeteaMask sends an `eth_call` that includes every account, not just the selected one, to the selected JSON-RPC provider.

### Error messages or log output

_No response_

### Version

10.16.2

### Build type

_No response_

### Browser

Brave

### Operating system

Windows

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## Explanation

* Adds ability for the user to opt-out of batching requests to get balances for all accounts in the wallet
  * Toggle under settings > Security & Privacy
  * Toggle on the "Set advanced privacy settings" post-wallet creation onboarding screen
* Unit tests

* Fixes #15169

## Screenshots/Screencaps

https://user-images.githubusercontent.com/13814744/204883244-d6b680ee-ac25-4a1c-ba8a-4073c24ef0d3.mp4

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by pedronfigueiredo — COMMENTED** [review]
github diff checker is a bit confused on this file, I simply reordered the render functions on this file and added a new one

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 1]
```diff
@@ -83,24 +85,26 @@ export default class SecurityTab extends PureComponent {
     );
   }
 
-  renderMetaMetricsOptIn() {
+  renderIncomingTransactionsOptIn() {
```
Comment (pedronfigueiredo) [inline ui/pages/settings/security-tab/security-tab.component.js:88]: github diff checker is a bit confused on this file, I simply reordered the render functions on this file and added a new one

**Review by brad-decker — DISMISSED** [review]
This looks good to me, tested locally and works well. Great work @pedronfigueiredo 🎉

**Review by jpuri — DISMISSED** [review]
LGTM 👍

**Comment by pedronfigueiredo** [comment]
Marking it as a draft until tomorrow to get the final copy.

**Comment by brad-decker** [comment]
@pedronfigueiredo by default we let contributors with merge access merge their own PRs, but sometimes they do get merged by others. Marking as draft is appropriate here due to incoming changes but you can also flag with DO-NOT-MERGE when necessary to avoid others merging fully approved PRs. Example: #12847

**Review by danjm — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]
🎉

**Pull request merged**